### DISCUSSION TITLE
Install a third module within another brand new module

### DISCUSSION THREAD

**Discussion by calebeaires** [nuxt/nuxt#18720] [Questions] [answered]
While building a Nuxt 3 Module, we export the `defineNuxtModule` as the bootstrap of the application. How to register another module to be used within this brand new module. I want use nuxt-icons && nuxt-i18n within this module I am creating. Is it made with hooks or inside the setup?

**Comment by wobsoriano** [comment, marked as answer]
Just access the options property and push those modules. Something like:

```js
export default defineNuxtModule({
  meta: {
    name: 'my-module',
  },
  setup (_options, nuxt) {
    nuxt.options.modules.push('nuxt-icons', 'nuxt-i18n')
  }
})
```

Best to check first if these modules are already installed before pushing them.

Also, check `installModule` function from `@nuxt/kit` if it helps you in some way.
